flang/test/Lower/intrinsic-procedures/size.f90

```diff
@@ -0,0 +1,39 @@
+! RUN: bbc -emit-fir %s -o - | FileCheck %s
+
+! CHECK-LABEL: func @_QPsize_test() {
+subroutine size_test()
+  real, dimension(1:10, -10:10) :: a
+  !CHECK-DAG: %[[sz0:.*]] = fir.alloca !fir.array<10x21xf32> {bindc_name = "a", uniq_name = "_QFsize_testEa"}
+  integer :: dim = 1
+  !CHECK-DAG: %[[sz1:.*]] = fir.address_of({{.*}}) : !fir.ref<i32>
+  integer :: iSize
+  !CHECK-DAG: %[[sz2:.*]] = fir.alloca i32 {bindc_name = "isize", uniq_name = "{{.*}}"}
+  iSize = size(a(2:5, -1:1), dim, 8)
+  !CHECK-DAG: %[[sz3:.*]] = fir.shape_shift %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}} : (index, index, index, index) -> !fir.shapeshift<2>
```

Comment:
Please do not use CHECK-DAG in this way.

Reply:
I'm not sure what you mean.  Are you referring to the use of "{{.*}}"?  I was using them to substitute for the names of constants.  What's the best practice here?

Reply:
In my latest commit I've changed most of the calls to be CHECK rather than CHECK-DAG.

Reply:
+1